packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+
+class AuthenticationSnippets implements DocSnippet {
+  @override
+  void runAll() {
+    getStarted_authStateChanges();
+    getStarted_idTokenChanges();
+    getStarted_userChanges();
+    manageUsers_getProfile();
+    passwordAuth_createPasswordBasedAccount();
+    emailLinkAuth_sendAuthLinkEmail();
+    social_google_sign_in();
+    social_twitterAuth();
+    social_facebookAuth();
+    phoneAuth_verifyPhoneNumber();
+    customAuth_withFirebase();
+    anonymousAuth_withFirebase();
+  }
+
+  void getStarted_authStateChanges() async {
+    // [START get_started_auth_state_changes]
+    FirebaseAuth.instance.authStateChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_auth_state_changes]
+  }
+
+  void getStarted_idTokenChanges() async {
+    // [START get_started_id_token_changes]
+    FirebaseAuth.instance.idTokenChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_id_token_changes]
+  }
+
+  void getStarted_userChanges() async {
+    // [START get_started_user_changes]
+    FirebaseAuth.instance.userChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
+      } else {
+        print('User is signed in!');
+      }
+    });
+    // [END get_started_user_changes]
+  }
+
+  void manageUsers_getProfile() async {
+    // [START manage_users_get_profile]
+    final user = FirebaseAuth.instance.currentUser;
+    if (user != null) {
+      // Name, email address, and profile photo URL
+      final name = user.displayName;
+      final email = user.email;
+      final photoUrl = user.photoURL;
+
+      // Check if user's email is verified
+      final emailVerified = user.emailVerified;
+
+      // The user's ID, unique to the Firebase project. Do NOT use this value to
+      // authenticate with your backend server, if you have one. Use
+      // User.getIdToken() instead.
+      final uid = user.uid;
+    }
+    // [END manage_users_get_profile]
+
+    // [START manage_users_get_provider_specific_profile]
+    if (user != null) {
+      for (final providerProfile in user.providerData) {
+        // ID of the provider (google.com, apple.cpm, etc.)
+        final provider = providerProfile.providerId;
+
+        // UID specific to the provider
+        final uid = providerProfile.uid;
+
+        // Name, email address, and profile photo URL
+        final name = providerProfile.displayName;
+        final emailAddress = providerProfile.email;
+        final profilePhoto = providerProfile.photoURL;
+      }
+    }
+    // [END manage_users_get_provider_specific_profile]
+
+    // [START manage_users_update_profile]
+    await user?.updateDisplayName("Jane Q. User");
```

Comment:
The corresponding [web snippet](https://firebase.google.com/docs/auth/web/manage-users#update_a_users_profile) shows how to listen for the completion/failure of this operation, so we should show that here too (either with `await`/`catch` or with `then`/`catchError`)